lib/task-queue/TaskRouter.js

```diff
@@ -81,6 +87,11 @@ class TaskRouter {
 
       taskCreationLock = true;
 
+      // Normalize the url if it's not canonical
+      if (!message.data.urlIsCanonical) {
+        message.data.url = normalizeUrl(message.data.url);
```

Comment:
We still want to normalize a canonical URL. They shouldn't change things semantically, but will take care of things like character casing, url encoding etc.